python/valis/db/queries.py

```diff
@@ -115,3 +116,50 @@ def cone_search(ra: Union[str, float], dec: Union[str, float],
         where(vizdb.SDSSidStacked.cone_search(ra, dec, radius,
                                               ra_col='ra_sdss_id',
                                               dec_col='dec_sdss_id'))
+
+
+def carton_program_list(name_type: str) -> peewee.ModelSelect:
+    """
+    Return a list of either all cartons or programs from targetdb
+
+    Parameters
+    ----------
+    name_type: str
+        Which type you are searching on, either 'carton' or 'program'
+
+    Returns
+    -------
+    list
+        list of either all cartons in programs sorted in alphabetical order
+    """
+    model = targetdb.Carton.select(peewee.fn.DISTINCT(getattr(targetdb.Carton, name_type)))
+    model_list = [getattr(m, name_type) for m in model]
+    model_list.sort()
+    return model_list
+
+
+def carton_program_search(name: str, name_type: str) -> peewee.ModelSelect:
+    """
+    Perform a search on either carton or program
+
+    Parameters
+    ----------
+    name: str
+        Either the carton name or the program name
+    name_type: str
+        Which type you are searching on, either 'carton' or 'program'
+
+    Returns
+    -------
+    peewee.ModelSelect
+        the ORM query
+    """
+    model = vizdb.SDSSidFlat.select(peewee.fn.DISTINCT(vizdb.SDSSidFlat.sdss_id))\
+                            .join(targetdb.Target,
+                                  on=(targetdb.Target.catalogid == vizdb.SDSSidFlat.catalogid))\
+                            .join(targetdb.CartonToTarget)\
+                            .join(targetdb.Carton)\
+                            .where(getattr(targetdb.Carton, name_type) == name)
+    model_stack = vizdb.SDSSidStacked.select()\
+                                     .join(model, on=(model.c.sdss_id == vizdb.SDSSidStacked.sdss_id))
```

Comment:
Is the stacked table preferred here?  Because we're returning a list of unique sdss_id targets?

Reply:
For this, I just chose the stacked as that seemed to be what was returned for the other queries. I honestly don't have a preference. If we were to use the SDSSidFlat, you are right we would just want this to be unique sdss_id targets (so add a distinct in there).

Reply:
I think I'm ok with returning the stacked here as the default.